plugins/module_utils/foreman_helper.py

```diff
@@ -1445,7 +1451,10 @@ def _flatten_entity(entity, foreman_spec):
             flat_name = spec.get('flat_name', key)
             property_type = spec.get('type', 'str')
             if property_type == 'entity':
-                result[flat_name] = value['id']
+                if not isinstance(value, NoEntity):
```

Comment:
```suggestion
                if value is not NoEntity:
```
And you can switch branches to avoid the `not`.